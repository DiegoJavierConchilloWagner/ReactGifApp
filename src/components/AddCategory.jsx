import React, { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleInputSubmit = (e) => {
        e.preventDefault();

        console.log('handleSubmit', inputValue);

        if( inputValue.trim().length > 2 ){
            setCategories( cats => [ inputValue, ...cats ]);
            setInputValue('')
        }

        
    }

    return (
        <form onSubmit={handleInputSubmit} >
            <p> { inputValue }</p>
            <input  
                type="text" 
                value={inputValue} 
                placeholder="Buscar meme..."
                onChange={ handleInputChange } 
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}