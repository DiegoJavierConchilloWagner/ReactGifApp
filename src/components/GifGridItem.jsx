import React from 'react'
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url }) => {

    return (
        <div className="gifCard animate__animated animate__bounceInUp">
            <img src={ url } alt={ title } />
            <p> { title } </p> 
        </div>

    )
}

GifGridItem.prototypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}