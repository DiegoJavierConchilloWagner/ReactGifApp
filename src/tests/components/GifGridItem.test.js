import React from 'react';

import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItems', () => {
    
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url}/> );

    test('Debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de tener un parrafo en el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title )

    });

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        // console.log( img.html() );
        // console.log( img.props() );
        // console.log( img.prop('src') );
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('Debe de tener animate__bounceInUp', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        // console.log( div.prop('className') );
        expect( className.includes('animate__bounceInUp') ).toBe( true );
    });

});
