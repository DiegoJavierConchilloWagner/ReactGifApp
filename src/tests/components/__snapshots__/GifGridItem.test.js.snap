// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItems Debe de mostrar el componente correctamente 1`] = `
<div
  className="gifCard animate__animated animate__bounceInUp"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
